lib/pbench/agent/tool_meister.py

```diff
@@ -1043,87 +1043,130 @@ def _send_client_status(self, status: str) -> int:
                 ret_val = 0
         return ret_val
 
-    def init_tools(self, data: dict[str, str]) -> int:
-        """init_tools - setup all registered tools which have data collectors.
-
-        The Tool Data Sink will be setting up the actual processes which
-        collect data from these tools.
+    def _create_tool_directory(self, directory: str) -> Path:
+        """Create a temporary tool directory suitable for persistent or
+        transient tool use.
 
         Arguments:
 
-            data: a dictionary of the arguments sent to the Tool Meister
+            directory:  the directory value provided by the "init" or "start"
+                        tools message.
+
+                        When the Tool Meister is run in a container the
+                        "directory" parameter will not map into its namespace,
+                        so we always consider containerized Tool Meisters as
+                        remote.
+
+        Returns a Path object for the created diretory.
+
+        Raises a ToolException if the given directory is local and can't be
+        resolved, or if a temporary directory can't be created.
 
-        Returns 0 on success, # of failures otherwise.
         """
-        # Name of the temporary tool data directory to use when invoking
-        # tools.  This is a local temporary directory when the Tool Meister is
-        # remote from the pbench controller.
-        if self._params.controller == self._params.hostname:
-            # This is the case when the Tool Meister instance is running on
-            # the same host as the controller.  We just use the directory
+        local_dir = Path(directory)
+        if self._params.controller == self._params.hostname and local_dir.exists():
+            # The Tool Meister instance is running on the same host as the
+            # controller (not in a container).  We just use the directory
             # given to us in the `start` message.
             try:
-                _dir = Path(data["directory"]).resolve(strict=True)
-            except Exception:
-                self.logger.exception(
-                    "Failed to access provided result directory, %s", data["directory"]
-                )
-                self._send_client_status("internal-error")
-                return 1
+                base_dir = local_dir.resolve(strict=True)
+            except Exception as exc:
```

Comment:
Given that `local_dir.exists()` is `True`, can `local_dir.resolve(strict=True)` ever raise an exception that we would want to catch?  (Or, can we dispense with the `try` block, here?)

Reply:
Is [this](https://www.pythontutorial.net/python-oop/python-raise-from/) your understanding of what `raise from` does?